src/utils/generateEntities.ts

```diff
@@ -1,105 +1,34 @@
-import { EntityType } from "../types";
 import yaml from "js-yaml";
 import { faker } from "@faker-js/faker";
-
-interface EntityMetadata {
-  name: string;
-  description: string;
-  annotations?: Record<string, string>;
-}
-
-interface EntitySpec {
-  type?: string;
-  lifecycle?: string;
-  owner: string;
-  definition?: string;
-  domain?: string;
-}
-
-interface Entity {
-  apiVersion: string;
-  kind: string;
-  metadata: EntityMetadata;
-  spec: EntitySpec;
-}
+import { generateEntity } from "../generators/registry";
+import { EntityKind, Entity } from "../types";
 
 export const generateEntities = (
-  entityType: EntityType,
+  entityKind: EntityKind,
   amount: number,
   owner?: string | null,
   annotations?: Record<string, string>
 ): string => {
   const entities: Entity[] = [];
 
   for (let i = 0; i < amount; i++) {
-    const randomDescription = faker.commerce.productDescription();
-    const entityName = `example-${faker.hacker.abbreviation().toLowerCase()}-${
-      i + 1
-    }`;
-    const randomType = ["website", "service", "library"];
+    const entityName = `example-${faker.hacker.abbreviation().toLowerCase()}-${i + 1}`;
 
-    const metadata: EntityMetadata = {
+    const entity = generateEntity(entityKind, {
       name: entityName,
-      description: randomDescription,
-    };
-
-    if (annotations && Object.keys(annotations).length > 0) {
-      metadata.annotations = {
-        ...annotations,
-      };
-    }
```

Comment:
Think we need something like this (dealers choice) so we don't end up with:

```yaml
metadata:
  name: example-adp-1
  description: Featuring Argon-enhanced technology, our Computer offers unparalleled aware performance
  annotations: {} # If none are provided lets just remove the annotations (or possibly default to one)
```

Reply:
This is fixed. We are now checking if the value is undefined and if not, has at least one key prior to adding annotations.